### PR TITLE
Custom fixed patient demographics

### PR DESCRIPTION
**Pull request by RobertScalfani**
This Pull Request adds the ability to input a file of Fixed Demographics Records to predetermine a population's demographic attributes.
It also allows for multiple fixed records to be input for each person so that real-world errors in health records between different providers can be simulated. For instance, one person's record may have the first name "Alex Neil" and a birth year of "2010" at one provider, and a different record may have the name "Alex" with an erroneous birth year of "2001."
View `src/test/resources/fixed_demographics/fixed_demographics_test.json` for an example file which includes 4 fixed patients, each of whom have 3 records, one for each of their providers. This file also showcases which attributes you can fix.

To run a test, use: `./run_synthea Colorado -f src/test/resources/fixed_demographics/fixed_demographics_test.json`. It is dependent on the input state being accurate because Synthea requires a valid city in a given state for each patient. This city will need to be pulled from one of the person's fixed records and cross-checked with Synthea's available cities for a state for validity.

This branch and PR is based off of the branch: `https://github.com/synthetichealth/synthea/tree/synthetic-denver`

Note: Code coverage is failing due to the way the population snapshot code was brought in from master, and it does not have any tests for it.

### CODE REVIEW

File: src/test/resources/fixed_demographics/fixed_demographics_test.json [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+        "PHONE_NUMBER": "9373705",
+        "ADDRESS_STREET1": "12439 Josephine St",
+        "ADDRESS_STREET2": "",
+        "ADDRESS_CITY": "Thornton",
+        "ADDRESS_STATE": "Colorado",
+        "ADDRESS_COUNTRY": "",
+        "ADDRESS_ZIPCODE": "80241-2744",
+        "PARENT1_SURNAME": "Fredric",
+        "PARENT1_GIVEN_NAME": "Kimberly Adrienne",
+        "PARENT1_EMAIL": "honeyanneristm@gmail.com"
```
Comment (dehall) [inline src/test/resources/fixed_demographics/fixed_demographics_test.json:23]: I am extremely curious about where the email addresses in this file came from, especially this one, but more importantly we shouldn't include any real email addresses (or any other real individual location or contact information for that matter) in the repo.  Can you change these to either an invalid domain or `@example.com` or `@example.org` , and confirm the other addresses and phone numbers don't point to something real?

Comment (RobertScalfani) [reply src/test/resources/fixed_demographics/fixed_demographics_test.json:23]: I believe all of the information in the file is completely fake, but they are from @eedrummer so he can confirm. I will also go ahead and update the domains to `@example.com`.

Comment (RobertScalfani) [reply src/test/resources/fixed_demographics/fixed_demographics_test.json]: I've updated all the phone numbers/emails in the test files to be clearly synthetic with repeating number patterns like `343434@example.com`.

File: src/main/java/org/mitre/synthea/engine/Generator.java [hunk 2]
```diff
@@ -340,14 +370,42 @@ public void run() {
       }
     }
     Exporter.runPostCompletionExports(this, exporterRuntimeOptions);
+    //Exporter.runPostCompletionExports(this);
```
Comment (eedrummer) [inline src/main/java/org/mitre/synthea/engine/Generator.java:373]: Please remove commented code

File: src/main/java/org/mitre/synthea/export/FhirR4.java [hunk 3]
```diff
@@ -768,6 +806,64 @@ private static BundleEntryComponent encounter(Person person, BundleEntryComponen
     return entry;
   }
 
+  private static BundleEntryComponent navigationalAssistance(
```
Comment (eedrummer) [inline src/main/java/org/mitre/synthea/export/FhirR4.java:809]: This should be removed. It has to do with some CODI stuff and isn't relevant to mainline Synthea.

File: src/main/java/org/mitre/synthea/export/FhirR4.java [hunk 4]
```diff
@@ -768,6 +806,64 @@ private static BundleEntryComponent encounter(Person person, BundleEntryComponen
...
+    Base64BinaryType b64 = new Base64BinaryType(docContent.getBytes());
+    attachment.setDataElement(b64);
+    DocumentReference.DocumentReferenceContentComponent drcc
+        = new DocumentReference.DocumentReferenceContentComponent();
+    drcc.setAttachment(attachment);
+    dr.addContent(drcc);
+    return newEntry(bundle, dr);
+  }
+
+  private static BundleEntryComponent wic(
```
Comment (eedrummer) [inline src/main/java/org/mitre/synthea/export/FhirR4.java:838]: This should also be removed, also CODI specific

File: src/main/java/org/mitre/synthea/helpers/Utilities.java [hunk 5]
```diff
@@ -310,6 +318,34 @@ public static final String readResource(String filename) throws IOException {
     return Resources.toString(url, Charsets.UTF_8);
   }
 
+  /**
+   * Load a fixed demographics patient record with seed demographics in json format:
+   * see src/main/resources/patient_template.json for a working example
+   * @param index index of the entry in the fixed demographic file.
+   * @return The patient record.
+   */
+   public static final Patient loadFixedDemographicPatient(int index) {
```
Comment (eedrummer) [inline src/main/java/org/mitre/synthea/helpers/Utilities.java:327]: This has a bunch of checkstyle issues that need to be addressed.

**Comment by eedrummer** [comment]
@dehall or @jawalonoski could one of you take one last look at this? I'm good with merging it, but I feel slightly gross about that since this contains a bunch of my code.

File: src/main/java/org/mitre/synthea/helpers/Utilities.java [hunk 6]
```diff
@@ -18,11 +21,15 @@
...
 
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.Patient;
+
 import org.mitre.synthea.engine.Logic;
 import org.mitre.synthea.engine.State;
 import org.mitre.synthea.world.concepts.HealthRecord.Code;
 
 public class Utilities {
+  private static final FhirContext ctx = FhirContext.forR4();
```
Comment (dehall) [inline src/main/java/org/mitre/synthea/helpers/Utilities.java:32]: Are this & the other imports in this file actually needed? It's private and I don't see any other changes below this in the file

**Review by dehall — APPROVED** [review]

**Pull request merged**